@kiva/kv-components/src/vue/KvSideSheet.vue

```diff
@@ -10,59 +10,93 @@
 		@click.self="closeSideSheet"
 	>
 		<div
-			class="tw-fixed tw-right-0 tw-transition-all tw-duration-300 tw-bg-white
-				tw-overflow-y-auto tw-p-2"
+			class="tw-fixed tw-right-0 tw-transition-all tw-duration-300 tw-bg-white tw-overflow-y-auto"
 			:class="{
 				'tw-w-0 tw-delay-200 tw-opacity-0': !open,
 				'lg:tw-w-1/2 tw-w-full tw-opacity-full': open,
+				'tw-h-full': $slots.controls,
 			}"
-			:style="modalStyles"
 		>
 			<div
-				class="tw-flex tw-justify-between tw-transition-opacity tw-duration-500 tw-delay-200"
-				:class="{
-					'tw-opacity-0': !open,
-					'tw-opacity-full': open,
-				}"
+				class="tw-relative tw-h-full"
+				:style="modalStyles"
 			>
-				<button
-					class="hover:tw-text-action-highlight"
-					@click="closeSideSheet"
+				<div
+					class="tw-flex tw-justify-between tw-transition-opacity tw-duration-500 tw-delay-200
+					tw-px-3 tw-py-2 tw-border-b tw-border-tertiary"
+					:class="{
+						'tw-opacity-0': !open,
+						'tw-opacity-full': open,
+					}"
 				>
-					<kv-material-icon
-						class="tw-w-3 tw-h-3"
-						:icon="mdiClose"
-					/>
-				</button>
-
-				<button
-					v-if="showGoToLink"
-					class="hover:tw-text-action-highlight"
-					@click="goToLink"
+					<div class="tw-flex tw-items-cente tw-gap-1.5">
+						<button
+							class="hover:tw-text-action-highlight tw-flex tw-items-center tw-justify-center"
+							@click="closeSideSheet"
+						>
+							<kv-material-icon
+								class="tw-w-3 tw-h-3"
+								:icon="mdiArrowLeft"
+							/>
+						</button>
+						<h2>
+							{{ headline }}
+						</h2>
+					</div>
+
+					<div class="tw-flex tw-items-cente tw-gap-1.5">
+						<button
+							v-if="showGoToLink"
+							class="hover:tw-text-action-highlight tw-flex tw-items-center tw-justify-center"
+							@click="goToLink"
+						>
+							<kv-material-icon
+								class="tw-w-3 tw-h-3"
+								:icon="mdiExportVariant"
+							/>
+						</button>
+						<button
+							class="hover:tw-text-action-highlight tw-flex tw-items-center tw-justify-center"
+							@click="closeSideSheet"
+						>
+							<kv-material-icon
+								class="tw-w-3 tw-h-3"
+								:icon="mdiClose"
+							/>
+						</button>
+					</div>
+				</div>
+				<div
+					class="tw-p-4 tw-overflow-y-auto tw-transition-opacity tw-duration-500 tw-delay-200
+						tw-overscroll-y-contain"
+					:class="{
+						'tw-opacity-0': !open,
+						'tw-opacity-full': open,
+					}"
 				>
-					<kv-material-icon
-						class="tw-w-3 tw-h-3"
-						:icon="mdiLaunch"
-					/>
-				</button>
-			</div>
-			<div
-				class="tw-p-4 tw-overflow-y-auto tw-transition-opacity tw-duration-500 tw-delay-200
-					tw-overscroll-y-contain"
-				:class="{
-					'tw-opacity-0': !open,
-					'tw-opacity-full': open,
-				}"
-			>
-				<slot></slot>
+					<slot></slot>
+				</div>
+				<div
+					v-if="$slots.controls"
```

Comment:
I tested smt similar in `KvCartModal` but didn't work passing from vue components.
To ensure testing in non storybook environment...